### ISSUE TITLE
Automate vCenter RBAC for limited operations user

### ISSUE THREAD

**Issue by dougm** [vmware/vic#4444]
As a developer, I want to validate that VIC functions properly when using a limited set of vCenter resources and
permissions.

Currently, VIC CI runs tests with ESX/vCenter unlimited administrator credentials.  Both CI and developers need a tool
to automate:

* User creation

* Role definition

* Assign user to role

* Apply role to resources (such as a datacenter or cluster)

The manual steps are here: https://github.com/vmware/vic/blob/08-docs/doc/user_doc/vic_installation/set_up_ops_user.md

This story does not include CI integration, that is tracked in issue  #3951

**Comment by andrewtchin** [comment]
Creating the SSO account by CLI requires using this http://www.virtuallyghetto.com/2015/05/vcenter-server-6-0-tidbits-part-9-creating-managing-sso-users-using-dir-cli.html

**Comment by dougm** [comment]
We can skip user creation for now and use the existing root user, for example: https://github.com/vmware/vic/blob/master/infra/machines/vcsa/create-cluster.sh#L124

**Comment by hickeng** [comment]
Getting this done in the Nightlies will at least catch some issues, such as #4899 and #4817 - both customer found issues in 1.1

**Comment by hickeng** [comment]
@dougm There is #1689 for configuring RBAC as part of `vic-machine create`. Is this specifically for a parallel out-of-band config?

**Comment by dougm** [comment]
@hickeng yes, this was for out-of-band (govc+bash), no changes to vic-machine.

**Comment by andrewtchin** [comment]
My existing branch for this is here https://github.com/andrewtchin/vic/tree/4444/ops-user

**Comment by hickeng** [comment]
It's not viable to complete this issue in 1.4.0 - at the very least the user creation part will not be done.
The remainder are done, but may need refinement.

**Comment by dougm** [comment]
Note that support for creating users via gomvomi just went in yesterday: https://github.com/vmware/govmomi/pull/1092

**Comment by hickeng** [comment]
I've updated this issue to be focused on the creation of a per-VCH ops-user now that the underlying support is available in govmomi.
There is little purpose in having this creation out-of-band if we can inline it and make it available to customers by default.
